pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -3185,3 +3209,48 @@ func Test_storeTaskSpec(t *testing.T) {
 		t.Fatalf(diff.PrintWantGot(d))
 	}
 }
+
+func TestGetResultsShouldFilterInternalTektonResultType(t *testing.T) {
```

Comment:
i'm a bit confused by this test case, if this is unrelated to timeouts maybe we can merge it separately?

Reply:
Indeed this test case is not directly related to timeouts but a consequence of the necessity for an `InternalTektonResultType` in order to digest timeout status messages.

Currently `TaskRunResultType`, `PipelineResourceResultType` and `UnknownResultType` exist as a `ResultType` for a `PipelineResourceResult` or `TaskRunResult` (which is a `PipelineResourceResult`). All results that conform to one of these `ResultTypes` are exposed after a taskrun has completed.

The `InternalTektonResultType` adds the ability to process a `PipelineResourceResult` internally without exposing it.
This test confirms that this `ResultType` is filtered from the external `PipelineResourceResults` and `TaskRunResults`.

Would you suggest I create an entirely separate PR that introduces the `InternalTektonResultType` then?

Reply:
See #3138